src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketPullRequestSCMHead.java

```diff
@@ -0,0 +1,68 @@
+package com.atlassian.bitbucket.jenkins.internal.scm;
+
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketPullRequest;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import jenkins.scm.api.SCMHeadOrigin;
+import jenkins.scm.api.mixin.ChangeRequestCheckoutStrategy;
+import jenkins.scm.api.mixin.ChangeRequestSCMHead2;
+import org.apache.commons.lang3.StringUtils;
+
+public class BitbucketPullRequestSCMHead extends BitbucketSCMHead implements ChangeRequestSCMHead2 {
+
+    public static final int PR_NAME_BRANCH_MAX_LENGTH = 20;
+    public static final String PR_NAME_TEMPLATE = "pr%s--%s--%s";
```

Comment:
The name affects the title of the job in the Pull Requests tab. It's a good question of whether we should generate a seperate job for a name change. Any thoughts @dkjellin @jaguilar-atl?

Reply:
> then will Jenkins generate a separate job for the new name, or will it continue the preexisting job as the pull request ID still matches?

Good point. I haven't tested it myself, but yes, I think it will generate a separate job as it uses the name as the unique identifier. I do agree that the Job should stay the same for a PR regardless of the target branch.

With that said, we should probably just use the PR id here to ensure that's the case @atl-dyon.

Thanks for raising this @KalleOlaviNiemitalo !